contrib/win32/win32compat/w32fd.c

```diff
@@ -556,17 +556,17 @@ w32_io_process_fd_flags(struct w32_io* pio, int flags)
 
 	shi_flags = (flags & FD_CLOEXEC) ? 0 : HANDLE_FLAG_INHERIT;
 
-	if (SetHandleInformation(WINHANDLE(pio), HANDLE_FLAG_INHERIT, shi_flags) == FALSE) {
-		/* 
-		 * Ignore if handle is not valid yet. It will not be valid for 
-		 * UF_UNIX sockets that are not connected yet 
-		 */
-		if (GetLastError() != ERROR_INVALID_HANDLE) {
-			debug3("fcntl - SetHandleInformation failed  %d, io:%p",
-				GetLastError(), pio);
-			errno = EOTHER;
-			return -1;
-		}
+	HANDLE h = WINHANDLE(pio);
+	
+	/*
+	* Ignore if handle is not valid yet. It will not be valid for
+	* UF_UNIX sockets that are not connected yet
+	*/
+	if (!IS_INVALID_HANDLE(h) && (SetHandleInformation(h, HANDLE_FLAG_INHERIT, shi_flags) == FALSE)) {
```

Comment:
Recommend adding a MACRO like this for readability
#define IS_VALID_HANDLE(h) (!IS_INVALID_HANDLE(h))